cluster-autoscaler/core/utils.go

```diff
@@ -254,7 +254,9 @@ func getNodeInfosForGroups(nodes []*apiv1.Node, nodeInfoCache map[string]*schedu
 	processNode := func(node *apiv1.Node) (bool, string, errors.AutoscalerError) {
 		nodeGroup, err := cloudProvider.NodeGroupForNode(node)
 		if err != nil {
-			return false, "", errors.ToAutoscalerError(errors.CloudProviderError, err)
+			klog.Warningf("Failed to get nodegroup for node %s: %v", node.Name, err)
+			metrics.RegisterError(errors.ToAutoscalerError(errors.CloudProviderError, err))
```

Comment:
I think the fix should go to GCP cloud provider. Where we have more control over which situation we should treat as an error and which not. I do not believe we want to treat any error from `NodeGroupForNode` as "not-found".

t seems to me that to implement the changed behavior we only need to make local changes to GceCache.GetMigForInstance, so we do not return an error if we spot the problematic instance. We should also add negative cache for such broken instances so we do not end up with sending to much API call listing MIGs instances.

Reply:
@losipiuk I've done the changes about the initial issue.
How would you like to do this negative cache ?
Do we return a custom error and remove the instance from the cache at this moment ?

Reply:
Thanks. I replied what I meant by the change in `cache.go`